src/NFreactions/reactions/DORreaction.cpp

```diff
@@ -232,52 +232,170 @@ void DORRxnClass::remove(Molecule *m, unsigned int reactantPos)
 	//if(DEBUG_MESSAGE)cout<<"finished removing"<<endl;
 }
 
+int DORRxnClass::checkForCollision(Molecule *m, MappingSet* ms, int rxnIndex){
+	
+	set<int> tempSet = m->getRxnListMappingSet(rxnIndex);
+	for(set<int>::iterator it= tempSet.begin();it!= tempSet.end(); ++it){
+		MappingSet* ms2 = reactantTree->getMappingSet(*it);
+		if(MappingSet::checkForEquality(ms,ms2)){
+			return *it;
+		}
+	}
+	return -1;
 
-bool DORRxnClass::tryToAdd(Molecule *m, unsigned int reactantPos) {
 
-	//if(DEBUG_MESSAGE)cout<<endl<<endl<<"adding molecule to DORRxnClass"<<endl;
-	//if(DEBUG_MESSAGE)m->printDetails();
+}
+
+bool DORRxnClass::tryToAdd(Molecule *m, unsigned int reactantPos) {
+	if(DEBUG_MESSAGE)cout<<endl<<endl<<"adding molecule to DORRxnClass"<<endl;
+	if(DEBUG_MESSAGE)m->printDetails();
 	if(reactantPos==(unsigned)this->DORreactantIndex) {
-	//	if(DEBUG_MESSAGE)cout<<" ... as a DOR"<<endl;
+		if(DEBUG_MESSAGE)cout<<" ... as a DOR "<<this->name<<endl;
 		//cout<<"RxnListMappingId: "<<m->getRxnListMappingId(m->getMoleculeType()->getRxnIndex(this,reactantPos))<<endl;
 
 		// handle the DOR reactant
 		int rxnIndex = m->getMoleculeType()->getRxnIndex(this,reactantPos);
 
-		//cout<<"trying to add to the tree:"<<endl;
-		//m->printDetails();
+		if(DEBUG_MESSAGE)cout<<"trying to add to the tree:"<<endl;
 
 		if(reactantTree->getHasClonedMappings()) {
-			if(m->getRxnListMappingId(rxnIndex)>=0) {
-				//cout<<"removing"<<endl;
+			while(m->getRxnListMappingId(rxnIndex)>=0) {
+				if(DEBUG_MESSAGE)cout<<"removing"<<endl;
 				reactantTree->removeMappingSet(m->getRxnListMappingId(rxnIndex));
-				m->setRxnListMappingId(rxnIndex,Molecule::NOT_IN_RXN);
+				m->deleteRxnListMappingId(rxnIndex,m->getRxnListMappingId(rxnIndex));
 			}
 		}
-
+		//JJT: keep a list containing those mapping sets that will be deleted
+		set<int> deleteMs = m->getRxnListMappingSet(rxnIndex);
+		symmetricMappingSet.clear();
 		if(m->getRxnListMappingId(rxnIndex)>=0) {
-			//if(DEBUG_MESSAGE)cout<<"was in the tree, so checking if we should remove"<<endl;
-			if(!reactantTemplates[reactantPos]->compare(m)) {
-				//if(DEBUG_MESSAGE)cout<<"removing..."<<endl;
-				reactantTree->removeMappingSet(m->getRxnListMappingId(rxnIndex));
-				m->setRxnListMappingId(rxnIndex,Molecule::NOT_IN_RXN);
-			} else {}
+
+			
+			
+			if(DEBUG_MESSAGE)cout<<"was in the tree, so checking if we should remove"<<endl;
+			ms=reactantTree->pushNextAvailableMappingSet();
+			comparisonResult = reactantTemplates[reactantPos]->compare(m,reactantTree,ms,false,&symmetricMappingSet);
+
+			if(!comparisonResult) {
+				reactantTree->removeMappingSet(ms->getId());
```

Comment:
We are not doing m->setRxnListMappingId(rxnIndex,Molecule::NOT_IN_RXN); anymore since at teh end of the routine we are deleting all mappigns that were not confirmed to still be  a valid reaction-molecule mapping. remember that now a  molecule has a set of  mappings to a reaction instead of it being a signle int.